### ISSUE TITLE
disco_l475_iot1: Multiple definitions of z_timer_cycle_get_32, etc.

### ISSUE THREAD

**Issue by pfalcon** [zephyrproject-rtos/zephyr#33342]
```
01:51:29.428 INFO    - 197/398 disco_l475_iot1           tests/subsys/power/power_mgmt/subsys.power.device_pm FAILED Build failure (build)
...
01:52:21.958 /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: libzephyr.a(cortex_m_systick.c.obj): in function `__NVIC_SetPriority':
01:52:21.958 /home/buildslave/workspace/zephyr-upstream/modules/hal/cmsis/CMSIS/Core/Include/core_cm4.h:1822: multiple definition of `z_clock_driver_init'; libzephyr.a(stm32_lptim_timer.c.obj):/home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/stm32_lptim_timer.c:82: first defined here
01:52:21.958 /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: libzephyr.a(cortex_m_systick.c.obj): in function `z_clock_set_timeout':
01:52:21.958 /home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/cortex_m_systick.c:168: multiple definition of `z_clock_set_timeout'; libzephyr.a(stm32_lptim_timer.c.obj):/home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/stm32_lptim_timer.c:202: first defined here
01:52:21.958 /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: libzephyr.a(cortex_m_systick.c.obj): in function `z_clock_elapsed':
01:52:21.958 /home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/cortex_m_systick.c:229: multiple definition of `z_clock_elapsed'; libzephyr.a(stm32_lptim_timer.c.obj):/home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/stm32_lptim_timer.c:272: first defined here
01:52:21.958 /home/buildslave/srv/toolchain/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld: libzephyr.a(cortex_m_systick.c.obj): in function `z_timer_cycle_get_32':
01:52:21.958 /home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/cortex_m_systick.c:242: multiple definition of `z_timer_cycle_get_32'; libzephyr.a(stm32_lptim_timer.c.obj):/home/buildslave/workspace/zephyr-upstream/zephyr/drivers/timer/stm32_lptim_timer.c:300: first defined here
```

Full build log: https://ci.linaro.org/job/zephyr-upstream/6078/PLATFORM=disco_l475_iot1,ZEPHYR_TOOLCHAIN_VARIANT=zephyr,label=docker-xenial-amd64-13/console

**Comment by FRASTM** [comment]
this should fix double definition #33285

**Comment by erwango** [comment]
@pfalcon thanks for the issue, a PR is submitted, I'm waiting for review...
